editor/src/dev/yourkit.clj

```diff
@@ -71,67 +85,134 @@
        (println "Wrote" snapshot-path#)
        result#)))
 
+;; -----------------------------------------------------------------------------
+;; YourKit utility functions
+;; -----------------------------------------------------------------------------
+
+(defmacro force-garbage-collection!
+  "Forces a garbage collection."
+  ([] `(force-garbage-collection! (controller)))
+  ([^Controller controller]
+   `(.forceGc ~(with-meta controller {:tag `Controller}))))
+
+(defmacro advance-allocation-generation!
+  "Advances object allocation generation number. Optionally, a string describing
+  the new generation can be supplied. Note that capturing an allocation snapshot
+  will always advance the generation number automatically."
+  ([] `(advance-allocation-generation! (controller)))
+  ([controller-or-description]
+   `(let [controller-or-description# ~controller-or-description]
+      (if (string? controller-or-description#)
+        (advance-allocation-generation! (controller) controller-or-description#)
+        (advance-allocation-generation! controller-or-description# nil))))
+  ([^Controller controller ^String description]
+   `(.advanceGeneration ~(with-meta controller {:tag `Controller}) ~description)))
+
+(defmacro start-telemetry!
+  "Starts collecting performance telemetry. If telemetry is already being
+  collected, the call has no effect."
+  ([] `(start-telemetry! (controller)))
+  ([^Controller controller]
+   `(.startTelemetry ~(with-meta controller {:tag `Controller}))))
+
+(defmacro stop-telemetry!
+  "Stops collecting performance telemetry. If telemetry is not being collected,
+  the call has no effect."
+  ([] `(stop-telemetry! (controller)))
+  ([^Controller controller]
+   `(.stopTelemetry ~(with-meta controller {:tag `Controller}))))
+
+(defmacro clear-telemetry!
+  "Clears telemetry. Note: when telemetry starts, collected telemetry is cleared
+  automatically, so you do not have to explicitly call this function."
+  ([] `(clear-telemetry! (controller)))
+  ([^Controller controller]
+   `(.resetTelemetry ~(with-meta controller {:tag `Controller}))))
+
```

Comment:
We use macros a lot in this file to minimize function call overhead when profiling.